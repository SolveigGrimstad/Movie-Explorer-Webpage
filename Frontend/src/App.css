
body{
  background-color: #393737;
}

.header {
  background-color: rgb(136, 136, 136);
  width: 100%;
  height: 10%;
  display: flex;
  flex-direction: column;
  font-size: calc(3px + 2vmin);
  color:#393737;
}





