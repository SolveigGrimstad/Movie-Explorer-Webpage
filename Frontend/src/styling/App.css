.searchbar {
  width: 100%;
  display: flex;
  flex-direction: row;
  margin: 1%;
  justify-content: center;
}

.searchbar_buttons {
  margin-top: 5%;
  justify-content: center;
}

.card {
  width: 15%;
}

.movie_container {
  display: flex;
  flex-direction: row;
  justify-content: space-around;
  flex-wrap: wrap;
}

Moviebox {
  padding: 20%;
}

.sidemenu {
  height: auto;
  width: 100%;
  margin-left: 10%;
  margin-right: 10%;
}

.filternav {
  width: 100%;
  margin-left: 5%;
  margin-right: 5%;
}

.pagination {
  text-align: center;
}

.homepageHeader {
  text-align: center;
  font-family: Impact, Haettenschweiler, "Arial Narrow Bold", sans-serif;
}

.slideshow {
  height: 1000px;
}

.divHeart {
  background-color: white;
  width: 38px;
  height: 30px;
}
