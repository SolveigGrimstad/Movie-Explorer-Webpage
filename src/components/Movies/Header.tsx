import React from 'react';

function Header() {

  

    return (
  
      <div>
        
      
        <h1>Filmbiblotek</h1>
      </div>
    );
  }
  
  export default Header;
  