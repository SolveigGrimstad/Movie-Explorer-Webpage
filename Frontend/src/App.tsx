import React from "react";
import "../src/styling/App.css";
import Header from "./components/header";
import Footer from "./components/footer";
//import Searchbar from "./components/searchbar";
import Content from "./components/content";
import Homepage from "./components/homepage";
import Movieinfo from "./components/movieinfo";
import "../src/styling/materialize.css";
import "../src/styling/materialize.min.css";
import { BrowserRouter as Router, Switch, Route, Link } from "react-router-dom";
import Moviebox from "./components/moviebox";
//import Moviebox from "./components/moviebox";

//redux
import { Dispatch } from "redux"
import { useSelector, shallowEqual, useDispatch } from "react-redux"




function App() {
/*
  const movies: readonly Imovie[] = useSelector(
    (state: MovieState) => state.movie,
    shallowEqual
  )

  const dispatch: Dispatch<any> = useDispatch()

  const saveMovie = React.useCallback(
    (movie: Imovie) => dispatch(searchMovie(movie)),
    [dispatch]
  )
  */
  return (
    <div className="App">
      <Router>
        <Header />

        <Switch>
          <Route path="/allmovies">
            <Content />
          </Route>

          <Route path="/movieinfo/">
            <Movieinfo />
          </Route>

          <Route path="/">
            <Homepage />
          </Route>
        </Switch>

        <Footer />
      </Router>
    </div>
  );
}

export default App;
