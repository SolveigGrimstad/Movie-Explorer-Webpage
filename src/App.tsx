import React, {useState, useEffect} from 'react';
import './App.css';
import Movies from './components/Movies/Movies';

const App: React.FC = () => {

  return (

    <div>

      <Movies/>
    

    </div>
  );
}

export default App;
