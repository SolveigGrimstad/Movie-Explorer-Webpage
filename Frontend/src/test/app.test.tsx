import enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import App from "../App";
import Content from "../components/content";
import Movieinfo from "../components/movieinfo";
import Header from "../components/header";
import Footer from "../components/footer";
import Homepage from "../components/homepage";
import React from "react";

enzyme.configure({ adapter: new Adapter() });

describe("App", () => {
  it("should contain Content", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<Content />)).toBe(true);
  });
  it("should contain Movieinfo", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<Movieinfo />)).toBe(true);
  });
  it("should contain Header", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<Header />)).toBe(true);
  });
  it("should contain Footer", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<Footer />)).toBe(true);
  });
  it("should contain Homepage", () => {
    const wrapper = shallow(<App />);
    expect(wrapper.contains(<Homepage />)).toBe(true);
  });
});
