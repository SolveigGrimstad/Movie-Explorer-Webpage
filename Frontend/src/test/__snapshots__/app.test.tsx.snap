// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders when there are no items 1`] = `
<div
  className="App"
>
  <div
    className="navbar-fixed"
  >
    <nav>
      <div
        className="nav-wrapper deep-purple"
      >
        <a
          className="center brand-logo"
          href="#"
        >
          MovieSelector
        </a>
        <ul
          className="hand-on-med-and-down"
          id="nav-mobile "
        >
          <li>
            <a
              href="/"
              onClick={[Function]}
            >
              Home
            </a>
          </li>
          <li>
            <a
              href="/allmovies"
              onClick={[Function]}
            >
              Movies
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
  <div>
    <h1>
      Homepage
    </h1>
  </div>
  <div>
    <footer
      className="page-footer deep-purple"
    >
      <div
        className="container "
      >
        <div
          className="row"
        >
          <div
            className="col l6 s12"
          >
            <h5
              className="white-text"
            >
              Footer Content
            </h5>
            <p
              className="grey-text text-lighten-4"
            >
              You can use rows and columns here to organize your footer content.
            </p>
          </div>
          <div
            className="col l4 offset-l2 s12"
          >
            <h5
              className="white-text"
            >
              Links
            </h5>
            <ul>
              <li>
                <a
                  className="grey-text text-lighten-3"
                  href="#!"
                >
                  Link 1
                </a>
              </li>
              <li>
                <a
                  className="grey-text text-lighten-3"
                  href="#!"
                >
                  Link 2
                </a>
              </li>
              <li>
                <a
                  className="grey-text text-lighten-3"
                  href="#!"
                >
                  Link 3
                </a>
              </li>
              <li>
                <a
                  className="grey-text text-lighten-3"
                  href="#!"
                >
                  Link 4
                </a>
              </li>
            </ul>
          </div>
        </div>
      </div>
      <div
        className="footer-copyright"
      >
        <div
          className="container"
        >
          Â© 2014 Copyright Text
          <a
            className="grey-text text-lighten-4 right"
            href="#!"
          >
            More Links
          </a>
        </div>
      </div>
    </footer>
  </div>
</div>
`;
